Author: Ken Zhu <guigenz@codeaurora.org>
Date:   Fri Apr 9 08:35:20 2021 -0700

    netifd: fix link up event causes interface down

    The interface down process should filter out link up
    event besides interface up and update event.

    Change-Id: I4194ea0625a16d98f82dc7aaa699fbb2a8c21bdd
    Signed-off-by: Ken Zhu <guigenz@codeaurora.org>

diff -Nur a/proto-shell.c netifd-2021-01-09-753c351b/proto-shell.c
--- a/proto-shell.c	2021-04-08 17:50:06.008898194 -0700
+++ netifd-2021-01-09-753c351b/proto-shell.c	2021-04-08 17:50:24.256630830 -0700
@@ -269,7 +269,7 @@
 	struct proto_shell_dependency *pdep;
 	struct proto_shell_state *state;
 
-	if (ev == IFEV_UP || ev == IFEV_UPDATE)
+	if (ev == IFEV_UP || ev == IFEV_UPDATE || ev == IFEV_LINK_UP)
 		return;
 
 	pdep = container_of(dep, struct proto_shell_dependency, dep);
